tools/@aws-cdk/cli-args-gen/lib/cli-type-gen.ts

```diff
@@ -129,9 +130,13 @@ function kebabToPascal(str: string): string {
     .join('');
 }
 
-function normalizeDefault(defaultValue: any): string {
-  if (typeof defaultValue === 'boolean' || typeof defaultValue === 'string') {
-    return String(defaultValue);
+function normalizeDefault(defaultValue: any, type: string): string {
+  const validDefaults = ['boolean', 'string', 'number', 'object'];
```

Comment:
We should make this an enum or type so we don't have to pass an any for defaultValue

Reply:
this is a good idea, but the value we are sending into `normalizeDefault` is already typed as an `any` and can include weird stuff like functions. the point here is to catch the more common values for defaults and send the rest to the catch-all. i will add inline comments if i end up keeping this function

Reply:
i changed this section of the code entirely, and i think it will be more to your liking.